unmarshal.go

```diff
@@ -82,6 +86,58 @@ func unmarshalField(field reflect.Value, tag string, prefix string) error {
 	return nil
 }
 
+// expandVariables replaces placeholders with actual environment variable values or defaults if not set.
+func expandVariables(value string, structPtr interface{}) string {
+	// Handle both ${var} and $var syntax
+	re := regexp.MustCompile(`\$\{([^}]+)\}|\$([A-Za-z_][A-Za-z0-9_]*)`)
```

Comment:
This regexp will be computed each time you call the method, it uses a lot of resources

Move it out the method, especially because you are using must compile. It would panic quickly if you update the regexp to something invalid, not when this feature is used